api/src/main/java/org/apache/iceberg/ManageSnapshots.java

```diff
@@ -80,4 +80,72 @@
    * wapId
    */
   ManageSnapshots cherrypick(long snapshotId);
+
+  /**
+   * Create a new branch where the snapshot with the given id will be the head of the branch.
```

Comment:
I think this can be more direct: "Create a new branch pointing to the given snapshot id"